src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.String.cs

```diff
@@ -8,6 +8,8 @@ namespace System.Text.Json
 {
     public sealed partial class Utf8JsonWriter
     {
+        private char _cachedHighSurrogate;
```

Comment:
This should cover split surrogate pairs, but shouldn't there also be provision for partial UTF-8 code points?